exec/java-exec/src/main/java/org/apache/drill/exec/store/easy/json/JSONRecordReader.java

```diff
@@ -253,11 +268,20 @@ private void updateRunningCount() {
     runningRecordCount += recordCount;
   }
 
+  public void setInputStream(InputStream in) {
+    this.inputStream = in;
+  }
+
   @Override
   public void close() throws Exception {
     if (stream != null) {
       stream.close();
       stream = null;
     }
+
+    if (inputStream != null) {
+      inputStream.close();
+      inputStream = null;
+    }
```

Comment:
Doesn't the Jackson parser close the input stream when the parser itself is closed? At least, that is what I thought I found in my own JSON tests. If not, we've got a pile of open HDFS streams that we've never closed.

Reply:
I'm not sure.  I wanted to make sure I closed anything that I opened.